### PR TITLE
RELATED: RAIL-3428 - Add plugin to workspace part 1

### PR DESCRIPTION
**Pull request by lupko**
-  Initial mini-refactors
-  Slight CLI remodeling
-  Added config parsing for add command

---

Supported PR commands:

| Command                  | Description            |
| ------------------------ | ---------------------- |
| `ok to test`             | Re-run standard checks |
| `extended test`          | BackstopJS tests       |
| `extended check sonar`   | SonarQube tests        |
| `extended check cypress` | Cypress E2E tests      |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5719/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7452/ : FAILURE in 1m 53s

File: tools/dashboard-plugin-template/package.json [hunk 1]
```diff
@@ -65,6 +65,7 @@
         "@babel/preset-env": "^7.7.2",
         "@babel/preset-react": "^7.7.2",
         "@gooddata/eslint-config": "^2.1.0",
+        "@gooddata/plugin-toolkit": "^8.7.0-alpha.50",
```
Comment (no23reason) [inline tools/dashboard-plugin-template/package.json:68]: Why is this needed?

File: tools/plugin-toolkit/src/addPluginCmd/actionConfig.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        if (isEmpty(username)) {
+            throw new InputValidationError(
+                "username",
+                "",
+                "Unable to determine username to use when logging into GoodData platform. Please make sure GDC_USERNAME env variable is set in your session or in the .env file",
+            );
+        }
+        if (isEmpty(password)) {
+            throw new InputValidationError(
+                "username",
```
Comment (no23reason) [inline tools/plugin-toolkit/src/addPluginCmd/actionConfig.ts:117]: Probably should be "password"

File: tools/dashboard-plugin-template/package.json (hunk 1 above)
Comment (lupko) [reply tools/dashboard-plugin-template/package.json:68]: to be part of the plugin project after bootstrap :)

the idea is that once user gets the plugin project (by using npx .. init) they have everything in that project to take things e2e. i'll eventually add additional scripts into the template so that user can do `npm run add-plugin` or `npm run use-plugin`.

File: tools/plugin-toolkit/src/addPluginCmd/actionConfig.ts (hunk 2 above)
Comment (lupko) [reply tools/plugin-toolkit/src/addPluginCmd/actionConfig.ts]: fixed

File: tools/dashboard-plugin-template/package.json (hunk 1 above)
Comment (no23reason) [reply tools/dashboard-plugin-template/package.json:68]: Oh right, makes sense!

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5722/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7455/ : FAILURE in 1m 32s

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1882

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1623/ : SUCCESS in 3m 32s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1903/ : SUCCESS in 17s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/700/ : SUCCESS in 17m 02s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7456/ : FAILURE in 1m 29s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5726/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7459/ : SUCCESS in 15m 39s

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1882

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1625/ : FAILURE in 3m 08s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1905/ : SUCCESS in 25s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/702/ : SUCCESS in 15m 56s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7461/ : SUCCESS in 20m 49s
